install/kubernetes/helm/istio/templates/_affinity.tpl

```diff
@@ -40,4 +40,56 @@
           - {{ $key }}
     {{- end }}
   {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{- define "podAntiAffinity" }}
+{{- if or .Values.podAntiAffinityLabelSelector .Values.podAntiAffinityTermLabelSelector}}
+  podAntiAffinity:
+    {{- if .Values.podAntiAffinityLabelSelector }}
+    requiredDuringSchedulingIgnoredDuringExecution:
+    {{- include "podAntiAffinityRequiredDuringScheduling" . }}
+    {{- end }}
+    {{- if or .Values.podAntiAffinityTermLabelSelector}}
+    preferredDuringSchedulingIgnoredDuringExecution:
+    {{- include "podAntiAffinityPreferredDuringScheduling" . }}
+    {{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "podAntiAffinityRequiredDuringScheduling" }}
+    {{- range $index, $item := .Values.podAntiAffinityLabelSelector }}
+    - labelSelector:
+        matchExpressions:
+        - key: {{ $item.key }}
+          operator: {{ $item.operator }}
+          {{- if $item.value }}
+          values:
+          {{- $vals := split "," $item.value }}
+          {{- range $i, $v := $vals }}
+          - {{ $v }}
+          {{- end }}
+          {{- end }}
+      topologyKey: {{ $item.topologyKey }}
+    {{- end }}
+{{- end }}
+
+{{- define "podAntiAffinityPreferredDuringScheduling" }}
+    {{- range $index, $item := .Values.podAntiAffinityTermLabelSelector }}
+    - podAffinityTerm:
+        labelSelector:
+          matchExpressions:
+          - key: {{ $item.key }}
+            operator: {{ $item.operator }}
+            {{- if .value }}
+            values:
+            {{- $vals := split "," $item.value }}
+            {{- range $i, $v := $vals }}
+            - {{ $v }}
+            {{- end }}
+            {{- end }}
+        topologyKey: {{ $item.topologyKey }}
+      {{- if $item.weight }}
+      weight: {{ $item.weight }}
```

Comment:
should we have a default weight 100?  any reason why user needs to configure weight other than 100?

Reply:
> The weight field in preferredDuringSchedulingIgnoredDuringExecution is in the range 1-100.

Seems that we can specify `weight` to multiple podAffinityTerm

Reply:
yes, i saw your example that you can configure weight 50 for one labelselector, and another 50 for another labelselector.  I understand the k8s pod anti-affnity platform can support all these config nobs, but the question is if we need to expose everyone of them to our users?  If we don't have a common scenario to use them on istio components, I'd rather not expose them.

Reply:
OK, got it.